Dockerfile

```diff
@@ -3,19 +3,39 @@ ARG COMMIT=""
 ARG VERSION=""
 ARG BUILDNUM=""
 
-# Build Geth in a stock Go builder container
-FROM golang:1.18-alpine as builder
+# Build libzkp dependency
+FROM scrolltech/go-rust-builder:go-1.19-rust-nightly-2022-12-10 as chef
+WORKDIR app
+
+FROM chef as planner
+COPY ./rollup/circuitcapacitychecker/libzkp/ .
+RUN cargo chef prepare --recipe-path recipe.json
+
+FROM chef as zkp-builder
+COPY ./rollup/circuitcapacitychecker/libzkp/rust-toolchain ./
+COPY --from=planner /app/recipe.json recipe.json
+RUN cargo chef cook --release --recipe-path recipe.json
 
-RUN apk add --no-cache gcc musl-dev linux-headers git
+COPY ./rollup/circuitcapacitychecker/libzkp .
+RUN cargo build --release
+RUN find ./ | grep libzktrie.so | xargs -I{} cp {} /app/target/release/
+
+# Build Geth in a stock Go builder container
+FROM scrolltech/go-rust-builder:go-1.19-rust-nightly-2022-12-10 as builder
 
 ADD . /go-ethereum
-RUN cd /go-ethereum && go run build/ci.go install ./cmd/geth
+COPY --from=zkp-builder /app/target/release/libzkp.so /usr/local/lib/
+COPY --from=zkp-builder /app/target/release/libzktrie.so /usr/local/lib/
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib/
+RUN cd /go-ethereum && env GO111MODULE=on go run build/ci.go install -buildtags circuit_capacity_checker ./cmd/geth
 
 # Pull Geth into a second stage deploy alpine container
-FROM alpine:latest
+FROM ubuntu:20.04
```

Comment:
Any reason we switched to an Ubuntu base image?

Reply:
we want to add more powerful support for rust, 

alpine alone doesn't support rust that well.  that's why we use `scrolltech/go-rust-builder` based on ubuntu

then we have to use `ubuntu` here because `scrolltech/go-rust-builder` is based on `ubuntu`